src/quest_manager/templates/quest_manager/submission.html

```diff
@@ -71,17 +71,19 @@ <h3 class="panel-title">Quest Submission Form</h3>
             {% include "quest_manager/buttons_approvals.html" %}
           {% endwith %}
         {% else %}
-          <a id='save-draft-btn' class='btn btn-default'>
-            <span id="draft-text">Save Draft</span>
-            <span id="draft-spin" style="display:none"><i class="fa fa-spinner fa-spin fa-fw"></i><span class="sr-only">Saving draft...</span></span>
-            <span id="draft-check" style="display:none" class="text-success"><i class="fa fa-check fa-fw"></i><span class="sr-only">Draft saved successfully</span></span>
-          </a>
-          {% if submission.is_approved %}
-            <input type='submit' name='comment' class='btn btn-primary' value='Add Comment' />
-          {% else %}
-            <input id='tour-submit-quest' type='submit' name='complete' class='btn btn-primary' value='Submit Quest for Completion' />
-            <a class="btn btn-danger" href="{% url 'quests:drop' submission.id %}" role="button">Drop Quest</a>
+          {% if submission.quest.visible_to_students %}
+            <a id='save-draft-btn' class='btn btn-default'>
+              <span id="draft-text">Save Draft</span>
+              <span id="draft-spin" style="display:none"><i class="fa fa-spinner fa-spin fa-fw"></i><span class="sr-only">Saving draft...</span></span>
+              <span id="draft-check" style="display:none" class="text-success"><i class="fa fa-check fa-fw"></i><span class="sr-only">Draft saved successfully</span></span>
+            </a>
+            {% if submission.is_approved %}
+              <input type='submit' name='comment' class='btn btn-primary' value='Add Comment' />
+            {% else %}
+              <input id='tour-submit-quest' type='submit' name='complete' class='btn btn-primary' value='Submit Quest for Completion' />
+            {% endif %}
           {% endif %}
+          <a class="btn btn-danger" href="{% url 'quests:drop' submission.id %}" role="button">Drop Quest</a>
```

Comment:
Why is this moved?  Won't this show students the "drop quests" button for quests that have already been approved?  They shouldn't be able to drop approved quests.

If my test suite wasn't so bad it would have caught this! haha

Reply:
Yeah. I forgot about it. I'll hide this from them and also change the queryset as well